src/emulator/apphosting/serve.ts

```diff
@@ -39,8 +39,9 @@ export async function start(options?: StartOptions): Promise<{ hostname: string;
 async function serve(port: number, startCommand?: string): Promise<void> {
   const rootDir = process.cwd();
   const apphostingLocalConfig = await getLocalAppHostingConfiguration(rootDir);
+
   const environmentVariablesToInject = {
-    ...apphostingLocalConfig.environmentVariables,
+    ...apphostingLocalConfig.environmentVariablesAsRecord(),
```

Comment:
why as Record?

Reply:
This seems like a unique enough use case that I would advocate for doing the mapping to Record type here as opposed to lifting it to the class definition.